Src/ILGPU.Algorithms/CL/CLMath.cs

```diff
@@ -0,0 +1,368 @@
+﻿// -----------------------------------------------------------------------------
+//                             ILGPU.Algorithms
+//                  Copyright (c) 2020 ILGPU Algorithms Project
+//                                www.ilgpu.net
+//
+// File: CLMath.cs
+//
+// This file is part of ILGPU and is distributed under the University of
+// Illinois Open Source License. See LICENSE.txt for details.
+// -----------------------------------------------------------------------------
+
+using ILGPU.Backends.OpenCL;
+using ILGPU.IR;
+using ILGPU.IR.Values;
+using ILGPU.Util;
+using System;
+using System.Runtime.CompilerServices;
+
+namespace ILGPU.Algorithms.CL
+{
+    static class CLMath
+    {
+        #region Code Generator
+
+        /// <summary>
+        /// Generates intrinsic math instructions for the following kinds:
+        /// Rcp
+        /// </summary>
+        /// <param name="backend">The current backend.</param>
+        /// <param name="codeGenerator">The code generator.</param>
+        /// <param name="value">The value to generate code for.</param>
+        public static void GenerateMathIntrinsic(
+            CLBackend backend,
+            CLCodeGenerator codeGenerator,
+            Value value)
+        {
+            var arithmeticValue = value as UnaryArithmeticValue;
+            var argument = codeGenerator.Load(arithmeticValue.Value);
+            var target = codeGenerator.Allocate(arithmeticValue);
+            var isFloat = arithmeticValue.BasicValueType.IsFloat();
+            if (arithmeticValue.Kind == UnaryArithmeticKind.RcpF && isFloat)
```

Comment:
I am wondering about the else case: Can this ever happen?

Reply:
You're right, else case is no longer needed - it used to be required because I had all the unary math functions listed.